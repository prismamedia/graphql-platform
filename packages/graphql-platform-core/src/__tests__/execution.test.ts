import { graphqlPlatform } from '.';
import complexMultiMutationRequest from './execution/complex-multi-mutation';
import complexMultiQueryRequest from './execution/complex-multi-query';

describe('Execution', () => {
  it('executes a complex multi query request', async done => {
    await expect(graphqlPlatform.execute(complexMultiQueryRequest)).resolves.toMatchSnapshot();

    done();
  });

  it('executes a complex multi mutation request', async done => {
    await expect(graphqlPlatform.execute(complexMultiMutationRequest)).resolves.toMatchSnapshot();

    done();
  });
});
