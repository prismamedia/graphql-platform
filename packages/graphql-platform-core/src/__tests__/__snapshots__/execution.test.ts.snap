// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Execution executes a complex multi mutation request 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: The "create" operation is not implemented, yet.],
  ],
}
`;

exports[`Execution executes a complex multi query request 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: The "find" operation is not implemented, yet.],
    [GraphQLError: The "find" operation is not implemented, yet.],
    [GraphQLError: The "find" operation is not implemented, yet.],
    [GraphQLError: The "find" operation is not implemented, yet.],
    [GraphQLError: The "find" operation is not implemented, yet.],
    [GraphQLError: The "find" operation is not implemented, yet.],
    [GraphQLError: The "count" operation is not implemented, yet.],
  ],
}
`;
